tools/msvc-build-launcher.cmd

```diff
@@ -1,39 +1,27 @@
-@echo off
-
-REM Use old Windows SDK 6.1 so created .exe will be compatible with
-REM old Windows versions.
-REM Windows SDK 6.1 may be downloaded at:
-REM  http://www.microsoft.com/en-us/download/details.aspx?id=11310
-set PATH_OLD=%PATH%
-
-REM The SDK creates a false install of Visual Studio at one of these locations
-set PATH=C:\Program Files\Microsoft Visual Studio 9.0\VC\bin;%PATH%
-set PATH=C:\Program Files (x86)\Microsoft Visual Studio 9.0\VC\bin;%PATH%
-
-REM set up the environment to compile to x86
-call VCVARS32
-if "%ERRORLEVEL%"=="0" (
-  cl /D "GUI=0" /D "WIN32_LEAN_AND_MEAN" launcher.c /O2 /link /MACHINE:x86 /SUBSYSTEM:CONSOLE /out:setuptools/cli-32.exe
-  cl /D "GUI=1" /D "WIN32_LEAN_AND_MEAN" launcher.c /O2 /link /MACHINE:x86 /SUBSYSTEM:WINDOWS /out:setuptools/gui-32.exe
-) else (
-  echo Windows SDK 6.1 not found to build Windows 32-bit version
-)
+del -R build
+mkdir build
+cd build
+call "cmake"
+if ERRORLEVEL 1 (echo Cannot locate CMake on PATH & exit /b 2)
+mkdir win32
+cd win32
+cmake -G "Visual Studio 17 2022" -A "Win32" ../../launcher
+MSBuild launcher.vcxproj /t:Build /property:Configuration=Release /property:Platform=Win32 /p:OutDir="../../setuptools" /p:TargetName="cli-32" /p:GUI=0
+MSBuild launcher.vcxproj /t:Build /property:Configuration=Release /property:Platform=Win32 /p:OutDir="../../setuptools" /p:TargetName="gui-32" /p:GUI=1
 
+cd ..
+cd ..
 REM buildout (and possibly other implementations) currently depend on
 REM the 32-bit launcher scripts without the -32 in the filename, so copy them
 REM there for now.
-copy setuptools/cli-32.exe setuptools/cli.exe
-copy setuptools/gui-32.exe setuptools/gui.exe
-
-REM now for 64-bit
-REM Use the x86_amd64 profile, which is the 32-bit cross compiler for amd64
-call VCVARSx86_amd64
-if "%ERRORLEVEL%"=="0" (
-  cl /D "GUI=0" /D "WIN32_LEAN_AND_MEAN" launcher.c /O2 /link /MACHINE:x64 /SUBSYSTEM:CONSOLE /out:setuptools/cli-64.exe
-  cl /D "GUI=1" /D "WIN32_LEAN_AND_MEAN" launcher.c /O2 /link /MACHINE:x64 /SUBSYSTEM:WINDOWS /out:setuptools/gui-64.exe
-) else (
-  echo Windows SDK 6.1 not found to build Windows 64-bit version
-)
+copy setuptools\cli-32.exe setuptools\cli.exe
+copy setuptools\gui-32.exe setuptools\gui.exe
 
-set PATH=%PATH_OLD%
+cd build
+mkdir 64
+cd 64
```

Comment:
Why the inconsistency in naming? Why not x64 to match the Platform indicated?

Reply:
refactored to python build script